File: Java/ProyectoIntegrador/Clases/Clases/Lista.java
package Clases;
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;

public abstract class Lista {

	public ArrayList<Object> lista;
	private File archivo;
	
	//Constructor
	public Lista(String name){
		lista = new ArrayList<Object>();
		archivo = new File(name);
		crearArchivo();
		leerArchivo();
	}
	
	//Archivos
	private void crearArchivo() {
		try {
			archivo.createNewFile();
		}catch (IOException e) {
			System.out.println("ERROR. LLAMA A SOPORTE TECNICO " +e.getMessage());
		}
	}
	
	protected void escribirArchivo() {
		try {
			FileWriter esc = new FileWriter(archivo);
				for(Object nodo: lista) {
					if(nodo instanceof Producto) esc.write(((Producto) nodo).txt());
					else if(nodo instanceof Venta) esc.write(((Venta)nodo).txt());
					else if(nodo instanceof Almacen) esc.write(((Almacen)nodo).txt());
				}
				esc.close();
			}catch(IOException e1) {
				System.out.println("ERROR AL SUBIR PRODUCTO" + e1.getMessage());
			}	
	}
	
	public void leerArchivo() {
		String linea = "";
		ArrayList<Object> objetos = new ArrayList<Object>();
		try {
			BufferedReader bf = new BufferedReader(new FileReader(archivo));
			while(linea != null) {
				linea = bf.readLine();
				if(linea != null) {
					String[] cadena = linea.split(",");
					if(cadena.length == 4) objetos.add(new Producto(cadena[0],cadena[1],cadena[2],cadena[3]));
					else if(cadena.length == 8) objetos.add(new Venta(cadena[0],cadena[1],cadena[2],cadena[3],cadena[4]));
					else if(cadena.length == 7) objetos.add(new Almacen(cadena[0],cadena[1],cadena[2],cadena[3],cadena[4],cadena[5],cadena[6]));
				}
			}
		lista = objetos;
		}catch(IOException e) {
			System.out.println(e.getMessage());
		}
	}
	
	//Listas
	
	public void agregar(Object nodo) {
		lista.add(nodo);
		this.escribirArchivo();
	}
	

	
	public abstract void mostrar();
	public abstract void eliminar(String id);
}

File: C#/ProyectoIntegradorCSHARP/ProyectoIntegradorCSHARP/Clases/Lista.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Collections;

namespace ProyectoIntegradorCSHARP.Clases
{
    abstract class Lista
    {
        public  ArrayList lista;
		private  String archivo;

		//Constructor
		public Lista(String name)
		{
			lista = new ArrayList();
			archivo = name;
			leerArchivo();
		}

		//Archivos


		protected void escribirArchivo()
		{
			try
			{
				TextWriter esc = new StreamWriter(archivo);
				foreach (Object nodo in lista)
				{
					if (nodo.GetType().Equals(typeof(Producto))) esc.WriteLine(((Producto)nodo).txt());
					else if (nodo.GetType().Equals(typeof(Venta))) esc.WriteLine(((Venta)nodo).txt());
					else if (nodo.GetType().Equals(typeof(Almacen))) esc.WriteLine(((Almacen)nodo).txt());
				}
				esc.Close();
			}
			catch(IOException e1) {
				Console.WriteLine("ERROR AL SUBIR PRODUCTO" + e1.Message);
	}
}

	public void leerArchivo()
	{
		String linea = "";
		ArrayList objetos = new ArrayList();
		try
		{
			TextReader bf = new StreamReader(archivo);
			while (linea != null)
			{
				linea = bf.ReadLine();
				if (linea != null)
				{
					String[] cadena = linea.Split(',');
					if (cadena.Length == 4) objetos.Add(new Producto(cadena[0], cadena[1], cadena[2], cadena[3]));
					else if (cadena.Length == 8) objetos.Add(new Venta(cadena[0], cadena[1], cadena[2], cadena[3], cadena[4]));
					else if (cadena.Length == 7) objetos.Add(new Almacen(cadena[0], cadena[1], cadena[2], cadena[3], cadena[4], cadena[5], cadena[6]));
				}
			}
			lista = objetos;
				bf.Close();
		}
		catch (Exception e)
		{
				Console.WriteLine(e.Message);
		}
	}

	//Listas

	public void agregar(Object nodo)
	{
		lista.Add(nodo);
		this.escribirArchivo();
	}



	public abstract void mostrar();
	public abstract void eliminar(String id);

    }
}
